main.py

```diff
@@ -70,7 +72,18 @@ def process_image(path: str, add_exif: bool, add_palette: bool, border_type: Bor
     if add_exif:
         exif = get_exif(img)
         if exif:
-            img_with_border = draw_exif(img_with_border, exif, border, fontname, boldfontname)
+            moduledir = os.path.dirname(os.path.abspath(__file__))
```

Comment:
Can you help me understand why this is outside the main function? When I did that, it threw errors but it may have been a result of my parameters eleswhere...

Reply:
I moved it to the exif if statement as we only want to do that processing if the user has requested exif to be added, and there is actual exif data to use. 

I'm not sure why it wouldn't work for you though, as the __file__ abspath should be resolved from within any function. 

This function is turning into a bit of spaghetti code though so at some point I'll have a go at refactoring it.